x-pack/legacy/plugins/apm/public/components/app/ServiceMap/Cytoscape.tsx

```diff
@@ -62,17 +62,30 @@ export function Cytoscape({
   // Trigger a custom "data" event when data changes
   useEffect(() => {
     if (cy) {
+      cy.remove(cy.nodes());
```

Comment:
Do we need to do the remove? As long as each node and edge have a unique `id` we should be able to add elements without removing. We might want to make this component use `memo` to only update if the list of ids changes.

Reply:
removed